### PR TITLE
GDPR: Privacy policy updates

### PR DESCRIPTION
**Pull request by orangejenny**
Opening for review. Won't be merged until the new documents are ready on dimagi.com and product team is ready (likely early April). Also need to verify whether or not this should affect enterprise servers.

@mkangia 

Spec: https://docs.google.com/document/d/1KoSgTVfK3hQnVAUkfLQMHQNUJZzShKdZaSth-eM3oIA/edit

Depends on https://github.com/dimagi/commcare-cloud/pull/1664

@dimagi/product following is a complete list of the changes made. These have been deployed to staging - I think it makes sense for y'all to test this rather than QA, but let me know if you'd also like a QA ticket.
- All users will need see a popup when they log into HQ that links to the new terms and does not allow them to proceed without agreeing.
- Links to the old EULA now point to the new docs:
   - When creating a new user (either via the usual method or via invitation)
   - HQ footer (single link to new docs instead of individual links to old EULA, privacy policy, and product subscription agreement)
- Miscellaneous text that refers to the EULA now references "Terms of Service" or "Terms of Service and Business Agreement":
   - Language in point 3 of the [content distribution agreement](https://www.commcarehq.org/exchange/cda/) - this is the one place that feels a little weird to update, since it's a legal doc
   - Error messages in exchange that come up if you don't agree to the terms
   - Language agreeing to be bound by the EULA when you transfer a domain
- Old product subscription agreement no longer exists, so you no longer need to check a box when:
   - Changing a project's plan (project settings > Current Subscription)
   - Confirming charges for extra users
- Links to [old privacy policy](https://www.dimagi.com/policy/) are replaced with link to new privacy doc:
   - When managing credit cards in billing
   - Next to the "Enable tracking" checkbox in my account settings that lets you opt out of analytics

In terms of reporting, we can look at any individual user and see the date they accepted the new terms (and can also see when/if they had accepted the old EULA).

### CODE REVIEW

**Review by mkangia — APPROVED** [review]

File: corehq/apps/accounting/templates/accounting/partials/new_stripe_card_template.html [hunk 1]
```diff
@@ -78,7 +78,7 @@ <h4 class="modal-title">{% trans "Add New Credit Card" %}</h4>
                                     <input type="checkbox"
                                            data-bind="checked: agreedToPrivacyPolicy, disable: isProcessing" />
                                     {% blocktrans %}
-                                        I agree to Dimagi's <a href="http://www.dimagi.com/policy" target="_blank">Privacy Policy</a>.
+                                        I agree to Dimagi's <a href="http://www.dimagi.com/terms/#privacy" target="_blank">Privacy Policy</a>.
```
Comment (biyeun) [inline corehq/apps/accounting/templates/accounting/partials/new_stripe_card_template.html]: fyi, the new policy link will be `https://www.dimagi.com/terms/latest/privacy/`

File: corehq/apps/accounting/templates/accounting/partials/stripe_card_ko_template.html [hunk 2]
```diff
@@ -163,7 +163,7 @@
                     <input type="checkbox"
                            data-bind="checked: agreedToPrivacyPolicy" />
                     {% blocktrans %}
-                        I agree to Dimagi's <a href="http://www.dimagi.com/policy" target="_blank">Privacy Policy</a>.
+                        I agree to Dimagi's <a href="http://www.dimagi.com/terms/#privacy" target="_blank">Privacy Policy</a>.
```
Comment (biyeun) [inline corehq/apps/accounting/templates/accounting/partials/stripe_card_ko_template.html]: `https://www.dimagi.com/terms/latest/privacy/`

File: corehq/apps/users/forms.py [hunk 3]
```diff
@@ -241,7 +241,7 @@ class UpdateMyAccountInfoForm(BaseUpdateUserForm, BaseUserInfoForm):
             "Allow Dimagi to collect usage information to improve CommCare. "
             "You can learn more about the information we collect and the ways "
             "we use it in our "
-            '<a href="http://www.dimagi.com/policy/">privacy policy</a>'
+            '<a href="http://www.dimagi.com/terms/#privacy">privacy policy</a>'
```
Comment (biyeun) [inline corehq/apps/users/forms.py]: `https://www.dimagi.com/terms/latest/privacy/`

File: corehq/apps/hqwebapp/templates/hqwebapp/base.html [hunk 4]
```diff
@@ -351,11 +351,9 @@
                                 <a href="{% trans 'http://www.commcarehq.org/home' %}">{% trans 'Learn more about CommCare HQ' %}</a>
```
Comment (biyeun) [inline corehq/apps/hqwebapp/templates/hqwebapp/base.html]: `commcarehq.org/home` no longer exists. should point to `dimagi.com/commcare`

**Review by biyeun — COMMENTED** [review]
have some tiny url edits

File: corehq/apps/accounting/templates/accounting/partials/new_stripe_card_template.html (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/accounting/templates/accounting/partials/new_stripe_card_template.html]: Should the links in this PR to `https://www.dimagi.com/terms` also be updated? Looks like that link now goes to the old privacy policy.

**Review by biyeun — APPROVED** [review]

**Review by biyeun — APPROVED** [review]

**Pull request merged**